### DISCUSSION TITLE
How do i give eye shape to barcode using this QRCoder ?

### DISCUSSION THREAD

**Discussion by GirishSolanki** [Shane32/QRCoder#311] [Q&A] [answered]
How do i give eye shape to barcode using this QRCoder ?

@marcelhintermann @pontusi @maartenba @nayuki  Can you please help me guys to sort it out into my applicaiton.

Thanks in Advance

**Comment by codebude** [comment, marked as answer]
> How do i give eye shape to barcode using this QRCoder ?
>

What do you mean by "eye shape"? Can you give an example?

**Edit:** See [last comment](https://github.com/codebude/QRCoder/discussions/311#discussioncomment-1690489) for solution.

**Reply by GirishSolanki** [reply]
LIke this 
![image](https://user-images.githubusercontent.com/20968896/128343356-74cfb385-9dc3-43dc-82b2-f5f926085131.png)

**Reply by codebude** [reply]
You can use the new [ArtQRCode](https://github.com/codebude/QRCoder/wiki/Advanced-usage---QR-Code-renderers#211-artqrcode-renderer-in-detail) renderer and set its `finderPatternImage`-paramter to render QR codes with your own finder pattern. (Just create an "eye shaped" finder pattern in the graphic editor of your choice and then pass it to the ArtQRCode render via the finderPatternImage parameter.)

**Reply by basharald** [reply]
Hi @codebude 

Would you please share a full sample of code for that? how we can use the finderpattern

**Comment by harshpandey0** [comment]
hey all i am new to this open source can anyone tell me which issues i have to pickup being as a beginner.

i would to love to connect with you all

LinkedIn : https://www.linkedin.com/in/harsh-pandey-2161hp
Skype : https://join.skype.com/invite/w7Fv2iEpdwlS
ShowCase : https://www.showwcase.com/harshpandey9079939
Twitter : https://twitter.com/harshpandey2121
Github : https://github.com/harsh2161

Thankyou
Harsh Pandey
[harshpandey9079@gmail.com](mailto:harshpandey9079@gmail.com)

**Comment by mahdiit** [comment]
> How do i give eye shape to barcode using this QRCoder ?
> 
> @marcelhintermann @pontusi @maartenba @nayuki Can you please help me guys to sort it out into my applicaiton.
> 
> Thanks in Advance

you can use Gufel.QrRender based on qrcoder with multiple option and also build high quality svg qrcode with 20 predefined template
see and use https://github.com/mahdiit/Gufel.QrRender
template list: https://github.com/mahdiit/Gufel.QrRender/blob/master/Asset/template.md
list of graphics : https://github.com/mahdiit/Gufel.QrRender/blob/master/Asset/graphics.md
